tensorboard/backend/application.py

```diff
@@ -451,14 +456,33 @@ def _serve_plugins_listing(self, request):
           A werkzeug.Response object.
         """
         response = collections.OrderedDict()
+        eid = plugin_util.experiment_id(request.environ)
+        plugins_with_data = (
+            self._data_provider.list_active_plugins(eid)
+            if self._data_provider is not None
+            else None
+        )
         for plugin in self._plugins:
             if (
                 type(plugin) is core_plugin.CorePlugin
             ):  # pylint: disable=unidiomatic-typecheck
                 # This plugin's existence is a backend implementation detail.
                 continue
             start = time.time()
-            is_active = plugin.is_active()
+
+            if plugins_with_data is not None:
+                is_active = (
+                    any(
```

Comment:
Optional, but IMO more obvious if we just check for a non-empty set intersection?